src/algebra/hom/centroid.lean

```diff
@@ -0,0 +1,239 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Christopher Hoskin
+-/
+import algebra.group.hom_instances
+import algebra.ring.basic
+
+/-!
+# Centroid homomorphisms
+
+This file defines centroid homomorphisms.
```

Comment:
Google doesn't give many results for "centroid homomorphism". Can you add some more background here, and maybe a reference?

Reply:
I've never heard the term "centroid homomorphism" before, I think @YaelDillies introduced it, presumably to comply with mathlib's naming conventions? I've added a reference to Jacobson.

Reply:
@YaelDillies, can you confirm that you made up this name? What's the closest terminology that you're used to, @mans0954?

Reply:
@eric-wieser the set of endomorphisms which commute with left and right multiplication is called the _centroid_. I don't recall hearing a term for the elements of the centroid  other than "let `T` be an element of the centroid of `A`".

Reply:
Can you adjust the docstring wording to start with the literature definition "centroid", and then explain "we call these `centroid_hom` to match `add_monoid_hom`" or similar?

Reply:
How's that https://github.com/leanprover-community/mathlib/pull/12746/commits/6824c47f2f2b82068b6c126b3dbfbb56919eaaf7 ?